core/spec/models/spree/concerns/user_address_book_spec.rb

```diff
@@ -398,5 +488,69 @@ module Spree
         expect(user.ship_address).to eq(address)
       end
     end
+
+    describe "bill_address=" do
+      let!(:user) { create(:user) }
+      let!(:address) { create(:address) }
+
+      # https://github.com/solidusio/solidus/issuesÎ/1241
+      it "resets the association and persists" do
+        # Load (which will cache) the has_one association
+        expect(user.bill_address).to be_nil
+
+        user.update!(bill_address: address)
+        expect(user.bill_address).to eq(address)
+
+        user.reload
+        expect(user.bill_address).to eq(address)
+      end
+    end
+
+    describe "#mark_default_address" do
+      let(:address) { build :address }
+
+      it "calls #mark_default_ship_address and warns user of deprecation" do
+        expect(user).to receive(:mark_default_ship_address)
+
+        expect {
+          user.mark_default_address(address)
+        }
+        .to output("Hey, this method is deprecated and it sets the ship_address only! Please start using #mark_default_ship_address for that\n").to_stdout
```

Comment:
Once you switch to `Spree::Deprecation`, you can still test this by using:

```ruby
expect(Spree::Deprecation).to receive(:warn)...
```

Great idea testing that we print this message anyway. 👏

Reply:
Done